mitiq/qse/tests/test_subspace_expansion.py

```diff
@@ -0,0 +1,210 @@
+    # Off diagonal terms are less than 1 because we are computing <Ψ|C_i C_j|Ψ>
+    # = <Ψ|C_k|Ψ> < 1 (since |Ψ> was rotated a bit from the logical subspace)
+    # All off diagonal terms are the same because of the symmetry of the
+    # total depolarizing noise.
+    off_diag_elements = S[np.where(~np.eye(16, dtype=bool))]
+    np.allclose(off_diag_elements, off_diag_elements[0])
+    assert off_diag_elements[0] < 1
+
+
+def test_compute_hamiltonian_overlap_matrix():
+    qc, _ = prepare_logical_0_state_for_5_1_3_code()
+    (
+        check_operators,
+        code_hamiltonian,
+    ) = get_check_operators_and_code_hamiltonian()
+
+    # If we have a full set of check operators that form a group then all
+    # entries of the H matrix should be the same.
+    # H_jk = sum over all i's <Ψ|C_i|Ψ>
+
+    H = _compute_hamiltonian_overlap_matrix(
+        qc, execute_no_noise, check_operators, code_hamiltonian, {}
+    )
+    assert np.allclose(H, -16 * np.ones(16))
+
+    H = _compute_hamiltonian_overlap_matrix(
+        qc,
+        execute_with_depolarized_noise,
+        check_operators,
+        code_hamiltonian,
+        {},
+    )
+    assert np.allclose(H, H[0][0])
+    assert H[0][0].real > -16
+
+
+def get_check_operators_and_code_hamiltonian() -> tuple:
+    Ms = [
+        "YIYXX",
+        "ZIZYY",
+        "IXZZX",
+        "ZXIXZ",
+        "YYZIZ",
+        "XYIYX",
+        "YZIZY",
+        "ZZXIX",
+        "XZZXI",
+        "ZYYZI",
+        "IYXXY",
+        "IZYYZ",
+        "YXXYI",
+        "XXYIY",
+        "XIXZZ",
+        "IIIII",
+    ]
+    Ms_as_pauliStrings = [
+        PauliString(M, coeff=1, support=range(5)) for M in Ms
+    ]
+    negative_Ms_as_pauliStrings = [
+        PauliString(M, coeff=-1, support=range(5)) for M in Ms
+    ]
+    Hc = Observable(*negative_Ms_as_pauliStrings)
+    return Ms_as_pauliStrings, Hc
+
+
+def prepare_logical_0_state_for_5_1_3_code():
+    def gram_schmidt(
+        orthogonal_vecs: List[np.ndarray],
+    ):
+        # normalize input
+        orthonormalVecs = [
+            vec / np.sqrt(np.vdot(vec, vec)) for vec in orthogonal_vecs
+        ]
+        dim = np.shape(orthogonal_vecs[0])[0]  # get dim of vector space
+        for i in range(dim - len(orthogonal_vecs)):
+            new_vec = np.zeros(dim)
+            new_vec[i] = 1  # construct ith basis vector
+            projs = sum(
+                [
+                    np.vdot(new_vec, cached_vec) * cached_vec
+                    for cached_vec in orthonormalVecs
+                ]
+            )  # sum of projections of new vec with all existing vecs
+            new_vec -= projs
+            orthonormalVecs.append(
+                new_vec / np.sqrt(np.vdot(new_vec, new_vec))
+            )
+        return orthonormalVecs
+
+    logical_0_state = np.zeros(32)
+    for z in ["00000", "10010", "01001", "10100", "01010", "00101"]:
+        logical_0_state[int(z, 2)] = 1 / 4
```

Comment:
Also in this case I trust you that this is correct, but some comments or a reference to understand how to derive the logical 0 state would be helpful.